### PR TITLE
✨ Support Flutter 3.0

### LINKED ISSUE #126: build fails with flutter beta
I've been using `pay: ^1.0.8` successfully on flutter's stable channel (`2.10.5`).

When I change to Flutter beta (`2.13.0-0.2.pre`), I get Android build errors.

```
e: /home/me/.pub-cache/hosted/pub.dartlang.org/pay_android-1.0.6/android/src/main/kotlin/io/flutter/plugins/pay_android/
      PayMethodCallHandler.kt: (50, 69): Type mismatch: inferred type is Activity? but Activity was expected
e: ...PayMethodCallHandler.kt: (69, 74): Type mismatch: inferred type is String? but String was expected
e: ...PayMethodCallHandler.kt: (73, 34): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Map<String, Any>?
e: .../PayMethodCallHandler.kt: (74, 34): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Map<String, Any>?
```

I can reproduce these errors building the example app (`pay/example/`, on branch `main`), after the following changes: 

```diff
diff --git a/pay/example/android/app/build.gradle b/pay/example/android/app/build.gradle
index 672a970..ba00dd9 100644
--- a/pay/example/android/app/build.gradle
+++ b/pay/example/android/app/build.gradle
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 31
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
@@ -40,7 +40,7 @@ android {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "io.flutter.plugins.pay_example"
         minSdkVersion 19
-        targetSdkVersion 28
+        targetSdkVersion 31
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }
diff --git a/pay/example/android/app/src/main/AndroidManifest.xml b/pay/example/android/app/src/main/AndroidManifest.xml
index 01892ab..6ac1174 100644
--- a/pay/example/android/app/src/main/AndroidManifest.xml
+++ b/pay/example/android/app/src/main/AndroidManifest.xml
@@ -1,11 +1,12 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="io.flutter.plugins.pay_example">
     <application
-        android:name="io.flutter.app.FlutterApplication"
+        android:name="${applicationName}"
         android:label="pay_example"
         android:icon="@mipmap/ic_launcher">
         <activity
             android:name="io.flutter.embedding.android.FlutterActivity"
+            android:exported="true"
             android:launchMode="singleTop"
             android:theme="@style/LaunchTheme"
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
diff --git a/pay/example/android/build.gradle b/pay/example/android/build.gradle
index 26859e8..1fe009b 100644
--- a/pay/example/android/build.gradle
+++ b/pay/example/android/build.gradle
@@ -1,12 +1,12 @@
 buildscript {
-    ext.kotlin_version = '1.4.20'
+    ext.kotlin_version = '1.6.10'
     repositories {
         google()
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
+        classpath 'com.android.tools.build:gradle:7.1.0'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
```

#### environment

```sh
$ flutter --version
Flutter 2.13.0-0.2.pre • channel beta • https://github.com/flutter/flutter
Framework • revision 8662e22bac (28 hours ago) • 2022-04-20 08:21:52 -0700
Engine • revision 24a02fa5ee
Tools • Dart 2.17.0 (build 2.17.0-266.5.beta) • DevTools 2.12.2
```

```
Android Studio Bumblebee | 2021.1.1 Patch 3
Build #AI-211.7628.21.2111.8309675, built on March 16, 2022
Runtime version: 11.0.11+0-b60-7590822 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.16.0-0.bpo.4-amd64
```

### PR DESCRIPTION
**Pull request by simonpham**
Closes #126

### CODE REVIEW

**Review by bilalsammour — APPROVED** [review]

**Review by pranavo72bex — APPROVED** [review]
👍

**Comment by kjawadDeveloper2** [comment]
Any update on when this will be merged.

**Comment by simonpham** [comment]
@JlUgia can you have a look at this?

**Comment by JlUgia** [comment]
Thank you @simonpham, taking a look.

File: pay/example/android/build.gradle [hunk 1]
```diff
@@ -1,20 +1,20 @@
 buildscript {
-    ext.kotlin_version = '1.4.20'
+    ext.kotlin_version = '1.6.10'
```
Comment (JlUgia) [inline pay/example/android/build.gradle:2]: Suggestion to update to `1.6.21` before accepting.

File: pay/example/android/build.gradle [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
+        classpath 'com.android.tools.build:gradle:7.1.2'
```
Comment (JlUgia) [inline pay/example/android/build.gradle:9]: Suggestion to update to `7.1.3` before accepting.

File: pay/example/android/app/build.gradle [hunk 3]
```diff
@@ -26,21 +26,29 @@ apply plugin: 'kotlin-android'
...
     }
 
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "io.flutter.plugins.pay_example"
+        // You can update the following values to match your application needs.
+        // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
         minSdkVersion 19
-        targetSdkVersion 28
+        targetSdkVersion flutter.targetSdkVersion
```
Comment (JlUgia) [inline pay/example/android/app/build.gradle:51]: This plugin still needs `minSdkVersion` to be `19`.

File: pay/example/android/app/build.gradle [hunk 4]
```diff
@@ -26,21 +26,29 @@ apply plugin: 'kotlin-android'
...
-    lintOptions {
-        disable 'InvalidPackage'
+    sourceSets {
+        main.java.srcDirs += 'src/main/kotlin'
     }
 
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "io.flutter.plugins.pay_example"
+        // You can update the following values to match your application needs.
```
Comment (JlUgia) [inline pay/example/android/app/build.gradle:48]: Let's add a new line above the comment to separate the two blocks.

File: pay_android/android/build.gradle [hunk 5]
```diff
@@ -18,30 +18,30 @@ group 'io.flutter.plugins.pay_android'
 version '1.0-SNAPSHOT'
 
 buildscript {
-    ext.kotlin_version = '1.4.20'
+    ext.kotlin_version = '1.6.10'
```
Comment (JlUgia) [inline pay_android/android/build.gradle:21]: Suggestion to update to 1.6.21 before accepting.

File: pay_android/android/build.gradle [hunk 6]
```diff
@@ -18,30 +18,30 @@ group 'io.flutter.plugins.pay_android'
...
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.1'
+        classpath 'com.android.tools.build:gradle:7.1.2'
```
Comment (JlUgia) [inline pay_android/android/build.gradle:28]: Suggestion to update to 7.1.3 before accepting.

File: pay_android/android/build.gradle [hunk 7]
```diff
@@ -18,30 +18,30 @@ group 'io.flutter.plugins.pay_android'
...
+        mavenCentral()
     }
 }
 
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 31
```
Comment (JlUgia) [inline pay_android/android/build.gradle:45]: I'd suggest we also add the 1_8 target options here:
```
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    kotlinOptions {
        jvmTarget = '1.8'
    }
```

**Comment by JlUgia** [comment]
Looks good, left a few quick comments.
Let me know when you can take a look and we are good to go.

File: pay/example/android/app/build.gradle (hunk 3 above)
Comment (JlUgia) [reply pay/example/android/app/build.gradle:51]: My bad

**Comment by simonpham** [comment]
@JlUgia done. Please check again.
Thank you for your time!

**Comment by JlUgia** [comment]
Looks good, merging.

**Comment by JlUgia** [comment]
`pay: 1.0.9` and `pay_android: 1.0.7` are now published.
Thank you for your contributions.

**Comment by pranavo72bex** [comment]
> `pay: 1.0.9` and `pay_android: 1.0.7` are now published. Thank you for your contributions.

Thanks bro 👍

**Pull request merged**